src/platform_impl/linux/event_loop.rs

```diff
@@ -462,35 +490,29 @@ impl<T: 'static> EventLoop<T> {
               glib::Propagation::Proceed
             });
             window.connect_button_press_event(move |window, event| {
-              if !window.is_decorated()
-                && window.is_resizable()
-                && !window.is_maximized()
```

Comment:
why was these checks removed?

Reply:
Interestingly, the behavior was the same in both Wayland and X11, regardless of whether the condition was present or not. Since the functionality was identical, I removed the condition.

Reply:
I would still keep them, I remember some window managers didn't enforce the same restrictions.

Reply:
Apologies, I made a mistake. After adding those conditions back and running it again, the conditions worked correctly. However, I’m confused about `!window.is_decorated()`. On macOS, it’s possible to resize from the window borders even without window decorations. 

Shouldn’t it be `window.is_decorated()` instead of `!window.is_decorated()` if we want to allow resizing when the window is decorated?

Reply:
macOS is the only OS that allows resizing even thought the window is without decorations but on Linux and Windows, we have to manually hit test and start resizing on our own.

`!window.is_decorated()` is the correct one, since we only want to do manual hit testing when the window is NOT decorated.

Reply:
I still don’t fully understand, but if that condition exists, resizing does not work on Linux (Wayland) when the window is decorated. Do we need a separate resize event for Wayland?

![resize-condition](https://github.com/user-attachments/assets/97ea14b4-da39-4f66-ae8c-2bbe8c5bfb30)

Reply:
we can just tweak the condition to always enable on wayland, `(is_wayland || window.is_decorated()) && window.is_resizable() && !window.is_maximized()`

Reply:
That condition seems to work well.
![drag](https://github.com/user-attachments/assets/75c2d604-c8f4-4311-9750-92412e6a3b14)